README.md

```diff
@@ -17,13 +37,112 @@ var completed = sourceList.filter(function(todo){
 });
 ```
 
-Any changes to `sourceList` will automatically update the derived `completed` list in O(log n) time.  
-This avoids the standard O(n) times done everywhere else.
+Any changes to `sourceList` will automatically update the derived `completed` list:
+```
+completed.bind('add', function(ev, newItems){
+	console.log(newItems);
+});
+
+sourceList.push({name: "cook", complete: true}, {name: "clean", complete: false});
+// `{name: "cook", complete: true}`
+```
+
+### With can.Map.define
+
+If you're using the [can.Map.define plugin](http://canjs.com/docs/can.Map.prototype.define.html), you can define a derived list like so:
+
+```
+{
+	define: {
+		todos: {
+			Value: can.List
+		},
+		completedTodos: {
+			value: function(){
+				return this.attr('todos').dFilter(function(todo){
+					return todo.attr('complete');
+				});
+			}
+		}
+	}
+}
+```
+
+Note that you are using `value` and not `get`. With `get`, every time `todos` changes, a brand new derived list will be generated. Since the derived list updates itself based on the source list, make sure only to instantiate the derived list once.
+
+## API
+
+The `can-derive` plugin adds the `dFilter` method to `can.List.prototype`, so all filtered lists are derived from a `can.List`.
+
+### dFilter
+
+`list.filter(predicate) -> DerivedList`
+
+Generates a derived list based on a predicate function.
+
+### attr
+
+`derivedList.attr() --> Array`
+
+Gets an array of all the elements in the derived list.
+
+`derivedList.attr(index) --> Object`
+
+Reads an element from an index on the filtered list. This references the *element in the source list* that *is at the specified index in the filtered list*.
+
+### each
+
+`derivedList.each(fn) --> DerivedList`
+
+Iterates through the DerivedList, calling a function for each element. The elements provided inside the function will be the elements *in the source list*. The filtered list will be updated only if the outcome of the predicate function changes.
+
+### Other can.List methods
+
+Since DerivedList is a [can.List](http://canjs.com/docs/can.List.html), the following methods are also available:
```

Comment:
DerivedList isn't a `can.List` exactly. It's a `can.RBTreeList`, which is meant to be `can.List` compliant.